dotCMS/src/main/java/com/dotmarketing/portlets/structure/action/EditFieldAction.java

```diff
@@ -552,4 +562,54 @@ public String hostNameUtil(ActionRequest req) {
         return h.getTitle()!=null?h.getTitle():"default";
     }
 
+    /**
+     * Validate if the radio field is valid
+     * @param fieldForm The field form
+     * @param req The action request
+     * @return true if the field is valid, false if not
+     */
+    private boolean isValidRadioField(FieldForm fieldForm, ActionRequest req){
+        boolean isValid = true;
+        final int EVEN_NUMBER = 2;
+        String dataType = fieldForm.getDataType();
+        String values = fieldForm.getValues();
+        String temp = values.replaceAll("\r\n","|");
+        String[] tempVals = StringUtil.split(temp.trim(), "|");
+
+        if (tempVals.length % EVEN_NUMBER == 1){
+            SessionMessages.add(req, "error", "message.structure.missingdatavalue");
+            return false;
+        } else if(dataType.equals(DataTypes.BOOL.toString())){
+            for(int i=1;i<tempVals.length;i+= 2){
+                if(!DbConnectionFactory.isDBBoolean(tempVals[i])){
+                    String message = "message.structure.invaliddataboolean";
+                    SessionMessages.add(req, "error", message);
+                    return false;
+                }
+            }
+        } else if(dataType.equals(DataTypes.FLOAT.toString())){
+            try {
+                for(int i=1;i<tempVals.length;i+= 2){
+                    Float.parseFloat(tempVals[i]);
+                }
+            }catch (NumberFormatException e) {
+                String message = "message.structure.invaliddata";
+                SessionMessages.add(req, "error", message);
+                return false;
+            }
+        } else if(dataType.equals(DataTypes.INTEGER.toString())){
+            try {
+                for(int i=1;i<tempVals.length;i+= 2) {
+                    Integer.parseInt(tempVals[i]);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The return value of "parseInt" must be used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2201)